errors/errors.go

```diff
@@ -1,22 +1,89 @@
-func (he *HTTPError) Error() string {
-	msg := ""
-	if he.Err != nil {
-		msg = he.Err.Error()
-	}
-	return fmt.Sprintf("%d: %s", he.Code, msg)
-}
+var (
+	// ErrNoStorage lint comment
+	ErrNoStorage = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "NO_STORAGE",
+		Message:        "The server is misconfigured and has no storage.",
+		Description:    "The server requires a storage backend to be configured to be capable of satisfying requests. This error is returned in the case that no such backend has been configured.",
+		HTTPStatusCode: http.StatusInternalServerError,
+	})
+	// ErrNoFilename lint comment
+	ErrNoFilename = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "NO_FILENAME",
+		Message:        "No file/role name provided.",
+		Description:    "An update was attempted without providing a file/role name to associate the data with.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrInvalidRole lint comment
+	ErrInvalidRole = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "INVALID_ROLE",
+		Message:        "The role you are attempting to operate on is invalid: %s",
+		Description:    "If a user attempts to perform an operation for a role that is not deemed valid, this error will be returned.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrMalformedJSON lint comment
+	ErrMalformedJSON = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "MALFORMED_JSON",
+		Message:        "JSON sent by a client could not be parsed by the server",
+		Description:    "This error indicates a client has sent malformed JSON.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrUpdating lint comment
+	ErrUpdating = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "UPDATING",
+		Message:        "An error has occurred while updating the TUF repository: %s",
+		Description:    "If an error occurs while attempting to update the storage with new TUF data, this error is returned.",
+		HTTPStatusCode: http.StatusInternalServerError,
+	})
+	// ErrMetadataNotFound lint comment
+	ErrMetadataNotFound = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "METADATA_NOT_FOUND",
+		Message:        "You have requested metadata that does not exist.",
+		Description:    "This error is returned when the user requests metadata that is not known to the server.",
+		HTTPStatusCode: http.StatusNotFound,
+	})
+	// ErrMalformedUpload lint comment
+	ErrMalformedUpload = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "MALFORMED_UPLOAD",
+		Message:        "The body of your request is malformed.",
+		Description:    "This error is raised when a user uploads new TUF data and we are unable to parse it as multipart/form-data.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrGenericNotFound lint comment
+	ErrGenericNotFound = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
That would probably be good. Re. my comments about the current Unknown, they should all take extra content to be output at the end of the message so we can specify what was not found, unknown, or unauthorized.